lib/Manager.php

```diff
@@ -224,6 +224,43 @@ public function resetAssignedSignalingServers(ICache $cache): void {
 		$result->closeCursor();
 	}
 
+	/**
+	 * @param string $searchToken
+	 * @param int $limit
+	 * @param int $offset
+	 * @return Room[]
+	 */
+	public function searchRoomsByToken(string $searchToken = '', int $limit = null, int $offset = null): array {
+		$query = $this->db->getQueryBuilder();
+		$query->select('*')
+			->from('talk_rooms')
+			->setMaxResults(1);
+
+		if ($searchToken !== '') {
+			$query->where($query->expr()->iLike('token', $query->createNamedParameter(
+				'%' . $this->db->escapeLikeParameter($searchToken) . '%'
+			)));
+		}
+
+		$query->setMaxResults($limit)
+			->setFirstResult($offset)
+			->orderBy('token', 'ASC');
+		$result = $query->execute();
+
+		$rooms = [];
+		while ($row = $result->fetch()) {
+			if ($row['token'] === null) {
+				// FIXME Temporary solution for the Talk6 release
```

Comment:
😨

Reply:
Nothing is more definitive than the temporary :laughing: No idea whether this is needed... I just copied it from `Manager::getRoomForParticipant()`. Is the check still necessary?